test/infrastructure/docker/templates/clusterclass-quick-start.yaml

```diff
@@ -32,6 +32,19 @@ spec:
             apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
             kind: DockerMachineTemplate
             name: quick-start-default-worker-machinetemplate
+    machinePools:
+    - class: default-mp-worker
+      template:
+        bootstrap:
+          ref:
+            apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
+            kind: KubeadmConfigTemplate
+            name: quick-start-default-worker-bootstraptemplate
+        infrastructure:
+          ref:
+            apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+            kind: DockerMachinePoolTemplate
+            name: quick-start-default-worker-machinepooltemplate
   variables:
   - name: imageRepository
     required: true
```

Comment:
Please ~ duplicate l.142-l.154 for MachinePools

The only difference should be:
* kind: DockerMachinePoolTemplate
* machineDeploymentClass => machinePoolClass
* path: "/spec/template/spec/template/customImage"
* template: |
            kindest/node:{{ .builtin.machinePool.version | replace "+" "_" }}

Reply:
This will give us nice e2e test coverage for inline JSON patching  (including builtin variables)